src/osd/PG.cc

```diff
@@ -4833,7 +4870,12 @@ void PG::start_peering_interval(
     dout(10) << __func__ << ": check_new_interval output: "
 	     << debug.str() << dendl;
     if (new_interval) {
-      dout(10) << " noting past " << past_intervals.rbegin()->second << dendl;
+      if ((osdmap->get_epoch() == osd->get_superblock().oldest_map) && (info.history.last_epoch_clean < osdmap->get_epoch())) {
+        dout(5) << "map gap, cleaning past_intervals and faking" << dendl;
+        past_intervals.clear();
+      } else {
+        dout(10) << " noting past " << past_intervals.rbegin()->second << dendl;
+      }
```

Comment:
hmm,I know clear past_intervals has fixed in luminous, but osdmap gap problem did not solve ; this PR is on hammer version, and I find that fix osdmap gap  need clear past_intervals,so I backport it